server/apps/main/tests/test_helpers.py

```diff
@@ -54,4 +62,215 @@ def test_moderator_check(self):
+        Test update PE in DB function
+        """
+        # data dict for creating PE
+        self.pe_data = {
+            'experience_text': "foo",
+            'difference_text': "bar",
+            'title_text': 'title',
+            'experience_id': 'foobar_id',
+            'viewable': True 
+        }
+        # assert no PE and PEH objects exist so far
+        self.assertEqual(len(PublicExperience.objects.all()),0)
+        self.assertEqual(len(ExperienceHistory.objects.all()),0)
+        # create new PE
+        update_public_experience_db(self.pe_data, "foobar_id" ,self.non_moderator_user)
+        # check that PE exists 
+        self.assertEqual(len(PublicExperience.objects.all()),1)
+        pe = PublicExperience.objects.get(experience_id='foobar_id')
+        # check that moderation status is "in review"
+        self.assertEqual(pe.moderation_status, 'not reviewed')
+        # check that this and only this PEH object exists 
+        self.assertEqual(len(ExperienceHistory.objects.filter(experience=pe)),1)
+        # check that EH object marks creation
+        peh = ExperienceHistory.objects.filter(experience=pe)[0]
+        self.assertEqual(peh.change_type,"Make Public")
+        # mark experience as moderation-approved
+        pe.moderation_status = 'approved'
+        pe.save()
+        # make new edit, should reset moderation status
+        self.pe_data['experience_id'] = 'foobar_id'
+        self.pe_data['viewable'] = True
+        update_public_experience_db(self.pe_data, "foobar_id" ,self.non_moderator_user)
+        # check moderation did reset properly
+        pe = PublicExperience.objects.get(experience_id='foobar_id')
+        self.assertEqual(pe.moderation_status, 'not reviewed')
+        # check that two history objects exists
+        self.assertEqual(len(ExperienceHistory.objects.filter(experience=pe)),2)
+        # get newest PEH object, check it's "Edit"
+        peh = ExperienceHistory.objects.all().order_by('changed_at')[1]
+        self.assertEqual(peh.change_type,"Edit")
+        # test moderatation edit
+        self.pe_data['viewable'] = True
+        self.pe_data['moderation_status'] = 'approved'
+        update_public_experience_db(self.pe_data, "foobar_id" ,self.moderator_user, 
+                                    change_type='Moderate',change_comments='I left a comment')
+        # check updated PE & latest PEH object
+        pe = PublicExperience.objects.get(experience_id='foobar_id')
+        self.assertEqual(pe.moderation_status, 'approved') # is now approved
+        self.assertEqual(len(ExperienceHistory.objects.filter(experience=pe)),3) # got 3 history entries now
+        peh = ExperienceHistory.objects.all().order_by('changed_at')[2] 
+        # check details of PEH & PE
+        self.assertEqual(pe.open_humans_member.oh_id,self.non_moderator_user.oh_id) # exp owned by owner
+        self.assertEqual(peh.changed_by.oh_id,self.moderator_user.oh_id) # exp last changed by moderator
```

Comment:
This test currently fails, but I think this is actually a mistake in how the function is written: 

The `update_public_experience_db` function currently always expects to be fed the _owner_ of the experience as the `ohmember` parameter. This is even true when a moderator changes the flag from `not reviewed` to `approved`. 

This is problematic, as the `update_public_experience_db` function uses the `ohmember` parameter to determine who made an action: https://github.com/alan-turing-institute/AutSPACEs/blob/bf004e2af946c663dff7226fc672aadcad94a362/server/apps/main/helpers.py#L261

As a consequence, any moderation action is currently stored as if it had been performed by the writer of the action. I think the easiest way to fix this would be to pass over the `request` parameter from the views into the helper function to fetch the user from there.

Reply:
I've fixed this in commit #d06fcd5